star_tides/core/actions/login_user_action.py

```diff
@@ -38,18 +35,19 @@ def __init__(self, username=None, password=None, token=None):
         self.password = password
         self.token = token
 
-    def run(self):
+    def run(self) -> (str, str):
         if self.username and self.password:
 
             password = self.password.encode('utf-8')
-            user = User.objects(email=self.username).first()
+            user = UserModel.objects(email=self.username).first()
 
             if user is None:
                 # TODO create a custom exception and raise it
                 print(f'{self.__class__.__name__} User not found')
 
             if bcrypt.checkpw(password, user.password):
-                return create_jwt(user.email)
+                # TODO create refresh token
+                return create_jwt(user.email), ''
             # TODO create a custom exception and raise it
             print(f'{self.__class__.__name__} Password not a match')
         elif self.token:
```

Comment:
You can restructure this code to do as follows in order to make it more readable:

```
if not self.token or not (self.username and self.password):
  raise BadAuthError(...)

if self.token:
  ... do stuff
   return ...

# We know username and password are present.
... do work for username and password